### ISSUE TITLE
[v2] Duplicate Definitions .../GoogleMobileAds/GADMobileAds.h

### ISSUE THREAD

**Issue by D1no** [invertase/react-native-firebase#189]
We build multiple variants of the iOS App. Oddly, our CI System complains now at random for one of the three builds, that their are "duplicate definitions". As we don't need admob, I just throw it out now. But leave the error reporting here for further investigation. Local builds work just fine. Incremental Builds are in generally something that should better not break, if it is related at all.

## Build Log
```bash
2085 ▸ Check Dependencies
2086 ▸ Compiling RNFirebaseAdMobRewardedVideo.m
2087 ▸ Compiling RNFirebaseAdMob.m
2088 » In file included from node_modules/react-native-firebase/ios/RNFirebase/admob/RNFirebaseAdMob.m:5:
2089 ✗ /privatenode_modules/react-native-firebase/ios/../../../ios/Pods/Headers/Public/Google-Mobile-Ads-SDK/GoogleMobileAds/GADMobileAds.h:[16,1] duplicate interface definition for class 'GADMobileAds'
2090 » @interface GADMobileAds : NSObject
2091 » ^
2092 ⚠ /privatenode_modules/react-native-firebase/ios/../../../ios/Pods/Headers/Public/Google-Mobile-Ads-SDK/GoogleMobileAds/GADMobileAds.h:[16,12] previous definition is here
2093 » @interface GADMobileAds : NSObject
2094 »            ^
2095 » In file included from node_modules/react-native-firebase/ios/RNFirebase/admob/RNFirebaseAdMob.m:5:
2096 ✗ /privatenode_modules/react-native-firebase/ios/../../../ios/Pods/Headers/Public/Google-Mobile-Ads-SDK/GoogleMobileAds/GADMobileAds.h:[38,36] property has a previous declaration
2097 » @property(nonatomic, assign) float applicationVolume;
2098 »                                    ^
2099 ⚠ /privatenode_modules/react-native-firebase/ios/../../../ios/Pods/Headers/Public/Google-Mobile-Ads-SDK/GoogleMobileAds/GADMobileAds.h:[38,36] property declared here
2100 » @property(nonatomic, assign) float applicationVolume;
2101 »                                    ^
2102 » In file included from node_modules/react-native-firebase/ios/RNFirebase/admob/RNFirebaseAdMob.m:5:
2103 ✗ /privatenode_modules/react-native-firebase/ios/../../../ios/Pods/Headers/Public/Google-Mobile-Ads-SDK/GoogleMobileAds/GADMobileAds.h:[43,35] property has a previous declaration
2104 » @property(nonatomic, assign) BOOL applicationMuted;
2105 »                                   ^
2106 ⚠ /privatenode_modules/react-native-firebase/ios/../../../ios/Pods/Headers/Public/Google-Mobile-Ads-SDK/GoogleMobileAds/GADMobileAds.h:[43,35] property declared here
2107 » @property(nonatomic, assign) BOOL applicationMuted;
2108 »                                   ^
2109 » In file included from node_modules/react-native-firebase/ios/RNFirebase/admob/RNFirebaseAdMob.m:5:
2110 ✗ /privatenode_modules/react-native-firebase/ios/../../../ios/Pods/Headers/Public/Google-Mobile-Ads-SDK/GoogleMobileAds/GADMobileAds.h:[46,62] property has a previous declaration
2111 » @property(nonatomic, readonly, strong) GADAudioVideoManager *audioVideoManager;
2112 »                                                              ^
2113 ⚠ /privatenode_modules/react-native-firebase/ios/../../../ios/Pods/Headers/Public/Google-Mobile-Ads-SDK/GoogleMobileAds/GADMobileAds.h:[46,62] property declared here
2114 » @property(nonatomic, readonly, strong) GADAudioVideoManager *audioVideoManager;
2115 »                                                              ^
2116 ⚠ node_modules/react-native-firebase/ios/RNFirebase/admob/RNFirebaseAdMobInterstitial.h:[15,20] pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Wnullability-completeness]
2117 » @property NSString *adUnitID;
2118 »                    ^
2119 ⚠ node_modules/react-native-firebase/ios/RNFirebase/admob/RNFirebaseAdMobInterstitial.h:[15,20] insert '_Nullable' if the pointer may be null
2120 » @property NSString *adUnitID;
2121 »                    ^
2122 ⚠ node_modules/react-native-firebase/ios/RNFirebase/admob/RNFirebaseAdMobInterstitial.h:[15,20] insert '_Nonnull' if the pointer should never be null
2123 » @property NSString *adUnitID;
2124 »                    ^
2125 ⚠ node_modules/react-native-firebase/ios/RNFirebase/admob/RNFirebaseAdMobRewardedVideo.h:[15,20] pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Wnullability-completeness]
2126 » @property NSString *adUnitID;
2127 »                    ^
2128 ⚠ node_modules/react-native-firebase/ios/RNFirebase/admob/RNFirebaseAdMobRewardedVideo.h:[15,20] insert '_Nullable' if the pointer may be null
2129 » @property NSString *adUnitID;
2130 »                    ^
2131 ⚠ node_modules/react-native-firebase/ios/RNFirebase/admob/RNFirebaseAdMobRewardedVideo.h:[15,20] insert '_Nonnull' if the pointer should never be null
2132 » @property NSString *adUnitID;
2133 »                    ^
2134 ** BUILD FAILED **
2135 The following build commands failed:
2136 	CompileC /tmp/sandbox/5921a19163d6920001205e26/bbbuild/Build/Intermediates/RNFirebase.build/ReleaseTester-iphoneos/RNFirebase.build/Objects-normal/armv7/RNFirebaseAdMob.o RNFirebase/admob/RNFirebaseAdMob.m normal armv7 objective-c com.apple.compilers.llvm.clang.1_0.compiler
2137 (1 failure)
2138

=== Checking for Custom Logs in buddybuild_artifacts ===

```

## BuddyBuild CI Catch
This is the result of a "rebuild without cache" operation.
![image](https://user-images.githubusercontent.com/2397125/27419659-741267fa-5722-11e7-920a-9c5f8962e2f0.png)

## Errors
![image](https://user-images.githubusercontent.com/2397125/27419707-a18eb3fa-5722-11e7-8c98-1b64d290c884.png)

![image](https://user-images.githubusercontent.com/2397125/27419714-a9716ec8-5722-11e7-82c1-2b99e8565aeb.png)

![image](https://user-images.githubusercontent.com/2397125/27419733-c18aa470-5722-11e7-8bc0-6107a4214aa2.png)

![image](https://user-images.githubusercontent.com/2397125/27419742-c991cd4c-5722-11e7-903b-4a854d8078e9.png)

**Comment by bitfabrikken** [comment]
Experiencing the same here.

**Comment by chrisbianca** [comment]
Strange... Are you referencing the Admob pod in your `Podfile`?  If you're not using Ads, then you don't need to, though that doesn't mean this isn't a valid problem.

@Ehesp have you seen this at all whilst you've been implementing it?

**Comment by D1no** [comment]
@chrisbianca sure I did  — commented out the pod now and all works fine.

```ruby
platform :ios, '9.0'

target 'shoplink' do
  # Uncomment the next line if you're using Swift or would like to use dynamic frameworks
  # use_frameworks!
  # Pods for 'shoplink'

  pod "Yoga", :path => "../node_modules/react-native/ReactCommon/yoga"
  pod 'React', :path => '../node_modules/react-native', :subspecs => [
    'BatchedBridge', # Required For React Native 0.45.0+
    'Core',
    # Add any other subspecs you want to use in your project
  ]

  # Required by RNFirebase
  pod 'Firebase/Core'
  pod 'RNFirebase', :path => '../node_modules/react-native-firebase'

  # [OPTIONAL PODS] - comment out pods for firebase products you won't be using.
  # pod 'Firebase/AdMob'
  pod 'Firebase/Analytics'
  pod 'Firebase/Auth'
  pod 'Firebase/Crash'
  pod 'Firebase/Database'
  pod 'Firebase/DynamicLinks'
  pod 'Firebase/Messaging'
  pod 'Firebase/RemoteConfig'
  pod 'Firebase/Storage'
end
```

![image](https://user-images.githubusercontent.com/2397125/27421756-06eb7af4-572c-11e7-9f89-29d16c474769.png)

**Comment by paulftw** [comment]
I'm experiencing the same error - compilation of RNFirebaseAdMob.m fails due to duplicate declaration of `@interface GADMobileAds`.
Starting build again and again after this error sometimes leads to successful build and deployment to a device.
A possible workaround is commenting out `#import "GoogleMobileAds/GADMobileAds.h` inside RNFirebaseAdMob.m. App would compile and run fine, interstitials and FCM would work as expected but banners won't load.

When I hit the rare and random successful compilation exactly same code results in working banners and interstitials.
XCode 8.3.3.
```
	"dependencies": {
		"react": "16.0.0-alpha.12",
		"react-native": "0.45.1",
		"react-native-code-push": "^3.0.1-beta",
		"react-native-fbsdk": "^0.6.1",
		"react-native-firebase": "^2.0.2"
	},
```
I know RN 0.46 is out already but it had a couple of babel bugs that couldn't find a workaround for.

**Comment by chrisbianca** [comment]
This is a really strange one as it doesn't seem to be deterministic.

@D1no @paulftw @bitfabrikken are these new projects? Or ones that you've started introducing RNFirebase too?  If the latter, had you previously been using Google Ads / Firebase Admob?  I'm wondering if there's something left from a previous installation.

**Comment by bitfabrikken** [comment]
I think what fixed it for me, was the Header List was full of weird entries. I just removed them all and put $inherited in 'em.

**Comment by paulftw** [comment]
@chrisbianca I started a new `react-native init` project and then followed the RNFirebase installation steps. It is indeed not deterministic. XCode and run-ios sometimes lead to different outcome. I suspect it even might be a race condition somewhere in XCode toolchain.

**Comment by chrisbianca** [comment]
@paulftw have you tried checking the header list as per @bitfabrikken suggestion?  This does sound more like an Xcode / iOS issue rather than a react-native-firebase issue, but I won't close quite yet.

**Comment by paulftw** [comment]
@chrisbianca @bitfabrikken what do you mean by the "Header List"? Ones I could find in XCode didn't have "weird entries".

**Comment by bitfabrikken** [comment]
@paulftw By weird I meant I had at one point 20+ entries in the header list (Build Settings > Header Search Paths > Double click). Reducing it to just $(inherited) fixed my issues.

I have had the duplicate error happen again today and yesterday. The way I fixed it was to do:

```
cd ios
rm -rf Pods
rm Podfile.lock
pod install
```

In one case, that did not resolve it, and I had to rm -rf the node_modules and do npm install again, then the above things, and it worked again.

Also remember to do Product > Clean in Xcode before building

**Comment by bitfabrikken** [comment]
I just tried a fresh install, new "react-native init MyApp" on a new mac, and I'm also (again) getting these duplicate errors + build failed, after installing react-native-firebase.

So it seems there's actually a bug here.

Snippet of package.json
```
        "dependencies": {
                "react": "16.0.0-alpha.12",
                "react-native": "0.46.3",
                "react-native-firebase": "^2.0.3",
                "react-native-linear-gradient": "^2.0.0",
                "react-native-localization": "^0.1.31",
                "react-native-version-number": "^0.1.2"
        },

```


Podfile:
```
# Uncomment the next line to define a global platform for your project
platform :ios, '8.0'

target 'MyApp' do
  # Uncomment the next line if you're using Swift or would like to use dynamic frameworks
  # use_frameworks!

  pod "Yoga", :path => "../node_modules/react-native/ReactCommon/yoga"
  pod 'React', :path => '../node_modules/react-native', :subspecs => [
    'BatchedBridge', # Required For React Native 0.45.0+
    'Core',
    # Add any other subspecs you want to use in your project
  ]


  # Pods for MyApp


  #pod 'React', :path => '../node_modules/react-native'
  pod 'BVLinearGradient', :path => '../node_modules/react-native-linear-gradient'


  pod 'ReactNativeLocalization', :path => '../node_modules/react-native-localization'


  # Required by RNFirebase
  pod 'Firebase/Core'
  pod 'RNFirebase', :path => '../node_modules/react-native-firebase'

  # [OPTIONAL PODS] - comment out pods for firebase products you won't be using.
  pod 'Firebase/AdMob'
  pod 'Firebase/Analytics'
  #pod 'Firebase/Auth'
  pod 'Firebase/Crash'
  #pod 'Firebase/Database'
  #pod 'Firebase/DynamicLinks'
  #pod 'Firebase/Messaging'
  #pod 'Firebase/RemoteConfig'
  #pod 'Firebase/Storage'




  target 'MyAppTests' do
    inherit! :search_paths
    # Pods for testing
  end

end

```


Header Search Paths in Xcode for target MyApp:
```
$(inherited) 
${PODS_ROOT}/Firebase/Core/Sources 
"${PODS_ROOT}/Headers/Public" 
"${PODS_ROOT}/Headers/Public/Firebase" 
"${PODS_ROOT}/Headers/Public/FirebaseAnalytics" 
"${PODS_ROOT}/Headers/Public/FirebaseCore" 
"${PODS_ROOT}/Headers/Public/FirebaseCrash" 
"${PODS_ROOT}/Headers/Public/FirebaseInstanceID" 
"${PODS_ROOT}/Headers/Public/Google-Mobile-Ads-SDK"
"${PODS_ROOT}/Headers/Public/GoogleToolboxForMac" "${PODS_ROOT}/Headers/Public/Protobuf"
"${PODS_ROOT}/Headers/Public/RNFirebase" 
"${PODS_ROOT}/Headers/Public/nanopb" 
$(SRCROOT)/../node_modules/react-native-firebase/ios/RNFirebase/** 
$(SRCROOT)/../node_modules/react-native-linear-gradient/BVLinearGradient 
$(SRCROOT)/../node_modules/react-native-localization 
$(SRCROOT)/../node_modules/react-native-version-number/ios
```



Errors when attempting to build:
```
Semantic Issue Group
/Users/me/react/MyApp/node_modules/react-native-firebase/ios/RNFirebase/storage/RNFirebaseStorage.m:459:17: Method 'moduleName' in protocol 'RCTBridgeModule' not implemented
/Users/me/react/MyApp/node_modules/react-native/React/Base/RCTBridgeModule.h:69:1: Method 'moduleName' declared here
/Users/me/react/MyApp/node_modules/react-native-firebase/ios/RNFirebase/auth/RNFirebaseAuth.m:835:17: Method 'moduleName' in protocol 'RCTBridgeModule' not implemented
/Users/me/react/MyApp/node_modules/react-native/React/Base/RCTBridgeModule.h:69:1: Method 'moduleName' declared here
/Users/me/react/MyApp/node_modules/react-native-firebase/ios/RNFirebase/config/RNFirebaseRemoteConfig.m:134:17: Method 'moduleName' in protocol 'RCTBridgeModule' not implemented
/Users/me/react/MyApp/node_modules/react-native/React/Base/RCTBridgeModule.h:69:1: Method 'moduleName' declared here
/Users/me/react/MyApp/node_modules/react-native-firebase/ios/RNFirebase/database/RNFirebaseDatabase.m:515:17: Method 'moduleName' in protocol 'RCTBridgeModule' not implemented
/Users/me/react/MyApp/node_modules/react-native/React/Base/RCTBridgeModule.h:69:1: Method 'moduleName' declared here

/Users/me/react/MyApp/ios/Pods/Google-Mobile-Ads-SDK/Frameworks/frameworks/GoogleMobileAds.framework/Versions/A/Headers/GADMobileAds.h:16:1: Duplicate interface definition for class 'GADMobileAds'
/Users/me/react/MyApp/node_modules/react-native-firebase/ios/RNFirebase/admob/RNFirebaseAdMob.m:5:9: In file included from /Users/me/react/MyApp/node_modules/react-native-firebase/ios/RNFirebase/admob/RNFirebaseAdMob.m:5:
/Users/me/react/MyApp/ios/Pods/Google-Mobile-Ads-SDK/Frameworks/frameworks/GoogleMobileAds.framework/Versions/A/Headers/GADMobileAds.h:16:12: Previous definition is here


/Users/me/react/MyApp/ios/Pods/Google-Mobile-Ads-SDK/Frameworks/frameworks/GoogleMobileAds.framework/Versions/A/Headers/GADMobileAds.h:38:36: Property has a previous declaration
/Users/me/react/MyApp/node_modules/react-native-firebase/ios/RNFirebase/admob/RNFirebaseAdMob.m:5:9: In file included from /Users/me/react/MyApp/node_modules/react-native-firebase/ios/RNFirebase/admob/RNFirebaseAdMob.m:5:
/Users/me/react/MyApp/ios/Pods/Google-Mobile-Ads-SDK/Frameworks/frameworks/GoogleMobileAds.framework/Versions/A/Headers/GADMobileAds.h:38:36: Property declared here

/Users/me/react/MyApp/ios/Pods/Google-Mobile-Ads-SDK/Frameworks/frameworks/GoogleMobileAds.framework/Versions/A/Headers/GADMobileAds.h:43:35: Property has a previous declaration
/Users/me/react/MyApp/node_modules/react-native-firebase/ios/RNFirebase/admob/RNFirebaseAdMob.m:5:9: In file included from /Users/me/react/MyApp/node_modules/react-native-firebase/ios/RNFirebase/admob/RNFirebaseAdMob.m:5:
/Users/me/react/MyApp/ios/Pods/Google-Mobile-Ads-SDK/Frameworks/frameworks/GoogleMobileAds.framework/Versions/A/Headers/GADMobileAds.h:43:35: Property declared here

/Users/me/react/MyApp/ios/Pods/Google-Mobile-Ads-SDK/Frameworks/frameworks/GoogleMobileAds.framework/Versions/A/Headers/GADMobileAds.h:46:62: Property has a previous declaration
/Users/me/react/MyApp/node_modules/react-native-firebase/ios/RNFirebase/admob/RNFirebaseAdMob.m:5:9: In file included from /Users/me/react/MyApp/node_modules/react-native-firebase/ios/RNFirebase/admob/RNFirebaseAdMob.m:5:
/Users/me/react/MyApp/ios/Pods/Google-Mobile-Ads-SDK/Frameworks/frameworks/GoogleMobileAds.framework/Versions/A/Headers/GADMobileAds.h:46:62: Property declared here
```

**Comment by paulftw** [comment]
@bitfabrikken I see this error almost every day and almost every time I manage to get past the compilation phase and unblock my project.
Here are my observation:

1. run-ios almost never works on my machine.
2. Building & running from XCode works eventually.
3. If you compile it once and don't touch xcworkspace / xcodeproject subsequent builds will reuse the existing object file and you will be able to continue development until you need to add a pod or whatever.
4. react-native run-ios prints out the command it uses to build - something that looks like `xcodebuild -workspace FooBar.xcworkspace -configuration Debug *yadayada*`. Sometimes when I do `cd ios && xcodebuild ...` in terminal xcodebuild succeeds and after that run-ios also starts to work.
5. I can compile and run the project if I comment out `#import GADMobileAds` in RNFirebaseAdMob.m. Once I've done that sometimes I can uncomment it back, hit build in XCode and it will work!
6. Wrapping content of GADMobileAds into oldschool `#ifndef #define <old content> #endif` helped me compile the project more than once.

Most of these sound like complete "WTF compilers don't work that way". It doesn't seem deterministic it all. And I haven't slept for the past 3 days fighting this and other compilation errors, so maybe I'm slowly going nuts :)

What I think is going on is GADMobileAds.h and friends aren't supposed to be included directly. Something causes XCode take different include/__has_include paths during compilation and when RNFirebase admob headers import GAD sub-modules and also each other. GADMobileAds gets included twice because it's `#import`-ed both directly in .m and indirectly from one of it's other headers.

Going thru every RNFirebase header file, removing every `#import GADSomething.h` and importing `GoogleMobileAds.h` on the as needed basis should in theory both work better and be a more correct way of using the Ads framework. But again, in theory most of the weirdness I've described above should never happen.

**Comment by bitfabrikken** [comment]
@paulftw uncommenting that line fixed it for me :)

**Comment by bitfabrikken** [comment]
@paulftw actually, it compiles fine now, but now it can't display admob banners. It says "Warning: Native component for "RNFirebaseAdMobBanner" does not exist.

**Comment by paulftw** [comment]
@bitfabrikken I have the same warning (no native component) on simulator but everything is ok when deployed to simulator.
And also cleaning build folder and rebuilding usually helps to get rid of the native component warning.

**Comment by bitfabrikken** [comment]
@paulftw the warning is within the app, and the admob banner can't be shown due to this

**Comment by paulftw** [comment]
@bitfabrikken right now both device and simulator display the banners just fine. I have no idea why.

**Comment by bitfabrikken** [comment]
@paulftw: can you share your package.json and Podfile? Curious what version of react etc. you're running.

**Comment by Ehesp** [comment]
The firebase docs recommend importing the entire of `@import GoogleMobileAds;`. Maybe it's better just to check for that. I'm not sure how XCode compiles dependancies tbh.

**Comment by bitfabrikken** [comment]
@Ehesp: I tried various ways to use that import, but I'm not knowledgable enough on it to get it working sadly.

**Comment by paulftw** [comment]
@bitfabrikken 
In my project I've edited the RNFirebaseAdMob.m file to not import GADMobileAds.

Just cleaned everything up and launched it in a simulator and on iPhone 6S+.
Banner ad and interstitial work as expected in both. Rewarded video works in simulator but on device AdMob returns "admob/error-code-no-fill". I think inventory issue is unrelated - my phone may have gotten throttled or blacklisted for testing on paid ads way too often or whatever.

My **Podfile**
```
platform :ios, '9.0'
# use_frameworks!

react_native_path = '../node_modules/react-native'

target 'GoldFever' do

  pod 'Yoga', :path => react_native_path + '/ReactCommon/yoga'

  pod 'React', :path => react_native_path, :subspecs => [
    'BatchedBridge',
    'Core',
  ]

  pod 'Firebase/Core'
  pod 'Firebase/AdMob'
  pod 'Firebase/Auth'
  pod 'Firebase/Analytics'
  pod 'Firebase/Crash'
  pod 'Firebase/Database'
  pod 'Firebase/DynamicLinks'
  pod 'Firebase/Messaging'
  pod 'Firebase/RemoteConfig'
  pod 'Firebase/Storage'

  pod 'FirebaseUI/Auth'
  pod 'FirebaseUI/Phone'
  pod 'FirebaseUI/Facebook'

  pod 'RNFirebase', :path => '../node_modules/react-native-firebase'
end
```

**package.json**
```
"dependencies": {
		"react": "16.0.0-alpha.12",
		"react-native": "0.45.1",
		"react-native-code-push": "^3.0.1-beta",
		"react-native-fbsdk": "^0.6.1",
		"react-native-firebase": "^2.0.3"
	},
	"devDependencies": {
		"babel-jest": "20.0.3",
		"babel-preset-react-native": "2.1.0",
		"jest": "20.0.4",
		"react-test-renderer": "16.0.0-alpha.12"
	},
```

**RNFirebaseAdMob.m**
```
IcanHazMacbook:GoldFever paul$ head -n 7 node_modules/react-native-firebase/ios/RNFirebase/admob/RNFirebaseAdMob.m
#import "RNFirebaseAdMob.h"

#if __has_include(<GoogleMobileAds/GADMobileAds.h>)

// #import "GoogleMobileAds/GADMobileAds.h"

#import "RNFirebaseAdMobInterstitial.h"
```

**Comment by Ehesp** [comment]
There's a `main.admob.js` I use for testing btw.

I'll have a play with this myself as well, I just assumed I could import the `GADMobileAds.h` like every other project tbh!

**Comment by bitfabrikken** [comment]
I just got it working (both compiling and showing the admob banner in app) by removing the if/then statement, and having "#import "GoogleMobileAds/GADMobileAds.h"" included

Full file that works for me now: https://gist.github.com/bitfabrikken/251252c6340cd5411f1c59597c1a8557

**Comment by chrisbianca** [comment]
@bitfabrikken glad you got it working, though unfortunately we wouldn't be able to adopt this as an approach in RNFirebase.  The reason for the if statements is to prevent the need for everybody to import the `Firebase/Admob` pod when using react-native-firebase.  We want to make sure that you're able to only include the firebase functionality that you want to use.

**Comment by bitfabrikken** [comment]
Just an update here.. 

If you do fresh "react-native init myapp" and follow the react-native-firebase (2.0.4) ios install instructions to a tee, you still get the below errors when building in Xcode. 

Of course you must enable "pod 'Firebase/AdMob'" in Podfile.

So there's definitely some bugginess going on.

```
/Users/bitfabrikken-mac/myapp/node_modules/react-native-firebase/ios/RNFirebase/auth/RNFirebaseAuth.m:835:17: Method 'moduleName' in protocol 'RCTBridgeModule' not implemented

/Users/bitfabrikken-mac/myapp/ios/Pods/Google-Mobile-Ads-SDK/Frameworks/frameworks/GoogleMobileAds.framework/Versions/A/Headers/GADMobileAds.h:16:1: Duplicate interface definition for class 'GADMobileAds'

/Users/bitfabrikken-mac/myapp/ios/Pods/Google-Mobile-Ads-SDK/Frameworks/frameworks/GoogleMobileAds.framework/Versions/A/Headers/GADMobileAds.h:38:36: Property has a previous declaration

/Users/bitfabrikken-mac/myapp/ios/Pods/Google-Mobile-Ads-SDK/Frameworks/frameworks/GoogleMobileAds.framework/Versions/A/Headers/GADMobileAds.h:43:35: Property has a previous declaration
```

**Comment by chrisbianca** [comment]
@bitfabrikken are these actually errors or warnings?

**Comment by bitfabrikken** [comment]
@chrisbianca errors

**Comment by bitfabrikken** [comment]
@chrisbianca for some reason the new project I made works now. Very odd. I notice there's no header search paths in the new project.

**Comment by D1no** [comment]
React-Native has some serious issues with the yarn and npm caches. From my experience, at any new install, you should run in the project folder:

`rm -rf ~/Library/Developer/Xcode/DerivedData && ps aux | grep node && watchman watch-del-all && git clean -fdx -e .vscode && yarn cache clean && npm cache clean --force && yarn install`

than start the packager once with

`npm run start -- --reset-cache`

The "runs on my machine" bug is quite eminent in rn.

**Comment by hoangdoan267** [comment]
@bitfabrikken My project worked normally at the first time. After one day, when I opened my project, i faced that issue. I try to replace an original to your file above but NativeExpress doesn't work.

**Comment by bitfabrikken** [comment]
I've revisited my project today, and upgraded to RN 0.47.1 and react-native-firebase 2.1.0.

The Duplicate Definitions errors are still occuring.

I've rm -rf'ed node_modules and reinstalled them with yarn, and done an 'pod install' within ios dir, clean build in xcode and such.

I used to be able to get it working by uncommenting the below, and it does compile with xcode if I do that. But now the ad itself won't show neither on mobile or devices; I just get a warning (yellow box) that says 

Now of course as I write this, it actually compiled and showed the ads. Only thing I did differenly was to manually delete the build dir under ios folder and remove all header search paths entries except for $(inherited). The paths were filled with all sorts of entries like $(SRCROOT)/../../React/**, $(SRCROOT)/../../react-native/React/**, etc.

Uncomment this in RNFirebaseAdMob.m: 
```#import "GoogleMobileAds/GADMobileAds.h"```
